src/types/UiaTextRangeBase.cpp

```diff
@@ -501,42 +527,41 @@ IFACEMETHODIMP UiaTextRangeBase::GetText(_In_ int maxLength, _Out_ BSTR* pRetVal
     // truncated.
     const bool getPartialText = maxLength != -1;
 
-    if (!_degenerate)
+    if (!IsDegenerate())
     {
         try
         {
-            const ScreenInfoRow startScreenInfoRow = _endpointToScreenInfoRow(_pData, _start);
-            const Column startColumn = _endpointToColumn(_pData, _start);
-            const ScreenInfoRow endScreenInfoRow = _endpointToScreenInfoRow(_pData, _end);
-            const Column endColumn = _endpointToColumn(_pData, _end);
-            const unsigned int totalRowsInRange = _rowCountInRange(_pData);
-            const TextBuffer& textBuffer = _pData->GetTextBuffer();
-
 #if defined(_DEBUG) && defined(UIATEXTRANGE_DEBUG_MSGS)
             std::wstringstream ss;
-            ss << L"---Initial span start=" << _start << L" and end=" << _end << L"\n";
-            ss << L"----Retrieving sr:" << startScreenInfoRow << L" sc:" << startColumn << L" er:" << endScreenInfoRow << L" ec:" << endColumn << L"\n";
+            ss << L"---Initial span start={" << _start.X << L", " << _start.Y << L"} and end={" << _end.X << ", " << _end.Y << L"}\n";
             OutputDebugString(ss.str().c_str());
 #endif
 
+            // if _end is at 0, we ignore that row because _end is exclusive
+            const auto& buffer = _pData->GetTextBuffer();
+            const unsigned int totalRowsInRange = (_end.X == buffer.GetSize().Left()) ?
+                                                      _end.Y - _start.Y :
+                                                      _end.Y - _start.Y + 1;
+            const auto lastRowInRange = _start.Y + totalRowsInRange - 1;
+
             ScreenInfoRow currentScreenInfoRow = 0;
             for (unsigned int i = 0; i < totalRowsInRange; ++i)
             {
-                currentScreenInfoRow = startScreenInfoRow + i;
-                const ROW& row = textBuffer.GetRowByOffset(currentScreenInfoRow);
+                currentScreenInfoRow = _start.Y + i;
+                const ROW& row = buffer.GetRowByOffset(currentScreenInfoRow);
                 if (row.GetCharRow().ContainsText())
                 {
                     const size_t rowRight = row.GetCharRow().MeasureRight();
                     size_t startIndex = 0;
                     size_t endIndex = rowRight;
-                    if (currentScreenInfoRow == startScreenInfoRow)
+                    if (currentScreenInfoRow == static_cast<ScreenInfoRow>(_start.Y))
                     {
-                        startIndex = startColumn;
+                        startIndex = _start.X;
                     }
-                    if (currentScreenInfoRow == endScreenInfoRow)
+                    else if (currentScreenInfoRow == static_cast<ScreenInfoRow>(_end.Y))

```

Comment:
```suggestion
                   if (currentScreenInfoRow == static_cast<ScreenInfoRow>(_end.Y))
```
This is why `GetText()` has been acting so stupid.

Reply:
Bingo!